src/subdomains/support/application/services/support.service.ts

```diff
@@ -69,31 +60,51 @@ export class SupportService {
     return this.transformResultArray(data, query.table);
   }
 
-  async getSupportData(userId: number): Promise<{
-    staking: {
-      deposits: Deposit[];
-      withdrawals: Withdrawal[];
-      rewards: Reward[];
-    };
-  }> {
-    const user = await this.userService.getUser(userId);
+  async getSupportData(query: SupportDataQuery): Promise<SupportReturnData> {
+    const user = await this.getUser(query);
     if (!user) throw new NotFoundException('User not found');
 
-    const deposits = await this.repos.deposit.getByUserId(userId);
-    const withdrawals = await this.repos.withdrawal.getByUserId(userId);
-    const rewards = await this.repos.reward.getByUserId(userId);
+    const deposits = await this.repos.deposit.getByUserId(user.id);
+    const withdrawals = await this.repos.withdrawal.getByUserId(user.id);
+    const rewards = await this.repos.reward.getByUserId(user.id);
 
     return {
-      staking: {
-        deposits,
-        withdrawals,
-        rewards,
-      },
+      user,
+      deposits,
+      withdrawals,
+      rewards,
     };
   }
 
   //*** HELPER METHODS ***//
 
+  private async getUser(query: SupportDataQuery): Promise<User> {
+    switch (query.table) {
+      case SupportTable.USER:
+        return this.userService.getUserByKey(query.key, query.value);
+      case SupportTable.WALLET:
+        return this.walletService.getWalletByKey(query.key, query.value).then((wallet) => wallet?.user);
+      case SupportTable.DEPOSIT:
+        return this.userService.getUser(
+          await this.depositService.getDepositByKey(query.key, query.value).then((deposit) => deposit?.staking.userId),
+        );
+      case SupportTable.WITHDRAWAL:
+        return this.userService.getUser(
+          await this.withdrawalService
+            .getWithdrawalByKey(query.key, query.value)
+            .then((withdrawal) => withdrawal?.staking.userId),
+        );
+      case SupportTable.REWARD:
+        return this.userService.getUser(
+          await this.rewardService.getRewardByKey(query.key, query.value).then((reward) => reward?.staking.userId),
+        );
+      case SupportTable.STAKING:
+        return this.userService.getUser(
+          await this.stakingService.getStakingByKey(query.key, query.value).then((staking) => staking?.userId),
+        );
+    }
+  }
```

Comment:
Maybe we should move all these methods to the respecting repositories and use the RepositoryFactory here? See e.g. `src/subdomains/monitoring/observers/staking-combined.observer.ts`.